canonicaljson.py

```diff
@@ -26,7 +26,10 @@ def _default(obj):
     if type(obj) is frozendict:
         # fishing the protected dict out of the object is a bit nasty,
         # but we don't really want the overhead of copying the dict.
-        return obj._dict
+        try:
+            return obj._dict
+        except AttributeError:
+            return dict(obj)
```

Comment:
I think this just wants a comment for rationale.

```suggestion
        except AttributeError:
            # When the C implementation of frozendict is used,
            # there isn't a `_dict` attribute with a protected dict
            # so we resort to making a copy of the dict
            return dict(obj)
```

Reply:
Note: since frozendict v1.2, the pure-python implementation of frozendict inherits from dict. In that situation there'd be no attribute `_dict`. (But we'd never hit this path, since the stdlib's JsonEncoder will treat any `dict` subclass as if it were a dict.)